### DISCUSSION TITLE
Building a PWA with Static Export

### DISCUSSION THREAD

**Discussion by docsteer** [vercel/next.js#72221] [Help]
### Summary

I am trying to build a progressive web application (PWA) for static export. I have tried both NextJS 15.0.3 and canary.

If I create an app using `next-create-app` with the default settings (app router, typescript), and then:

- Add `output: 'export'` to the config
- Add a `manifest.ts` file as described [here](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating)

I get a build error:

> `Error: export const dynamic = "force-static"/export const revalidate not configured on route "/manifest.webmanifest" with "output: export". See more info here: https://nextjs.org/docs/advanced-features/static-html-export`

As far as I can see from the documentation PWA is supported by static builds - am I missing something?

Thanks for any help!

### Additional information

```js
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 16017
  Available CPU cores: 12
Binaries:
  Node: 20.10.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.3-canary.4 // Latest available version is detected (15.0.3-canary.4).
  eslint-config-next: 15.0.3-canary.4
  react: 19.0.0-rc-603e6108-20241029
  react-dom: 19.0.0-rc-603e6108-20241029
  typescript: 5.6.3
Next.js Config:
  output: export
```

**Comment by icyJoseph** [comment]
Show us the manifest file.

**Comment by docsteer** [comment]
I used the one from the documentation here: https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps#1-creating-the-web-app-manifest (and added some icons in the described locations)
      
      import type { MetadataRoute } from 'next'
       
      export default function manifest(): MetadataRoute.Manifest {
        return {
          name: 'Next.js PWA',
          short_name: 'NextPWA',
          description: 'A Progressive Web App built with Next.js',
          start_url: '/',
          display: 'standalone',
          background_color: '#ffffff',
          theme_color: '#000000',
          icons: [
            {
              src: '/icon-192x192.png',
              sizes: '192x192',
              type: 'image/png',
            },
            {
              src: '/icon-512x512.png',
              sizes: '512x512',
              type: 'image/png',
            },
          ],
        }
      }

**Reply by icyJoseph** [reply]
Could you add an export const dynamic, set to force-static? As the error seems to suggest ~ shouldn't be necessary though, but let's see

**Reply by rmsy** [reply]
For what it's worth, that does not seem to resolve this issue for me (tried both with and without `dynamic = "force-static"`):

```
 ⨯ Error: export const dynamic = "force-static"/export const revalidate not configured on route "/manifest.webmanifest" with "output: export". See more info here: https://nextjs.org/docs/advanced-features/static-html-export
    at Object.<anonymous> (.next/server/app/manifest.webmanifest/route.js:5:26) {
  page: '/manifest.webmanifest'
}
```

**Comment by rodrigozx** [comment]
add this export to manifest.ts:

`import type { MetadataRoute } from "next";

export const dynamic = "force-static";

export default function manifest(): MetadataRoute.Manifest{....`

**Reply by rmsy** [reply]
Based on the reactions I assume your suggestion has worked for others, however I seem to still be encountering this regardless of whether or not I configure `force-static`:

```
 ⨯ Error: export const dynamic = "force-static"/export const revalidate not configured on route "/manifest.webmanifest" with "output: export". See more info here: https://nextjs.org/docs/advanced-features/static-html-export
    at Object.<anonymous> (.next/server/app/manifest.webmanifest/route.js:5:26) {
  page: '/manifest.webmanifest'
}
```

**Reply by Ecostack** [reply]
Same story for me, does not seem to work

**Comment by zdarin** [comment]
To Fix add to this files if have `page.tsx / layout.tsx / robots.ts / sitemap.ts / manifest.ts / not-found.tsx / route.ts`
on top `export const dynamic = "force-static";` then error will disappear i had same error on nextjs 15 now fixed
[https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic](nextjs documentation)

**Comment by abhigyantrips** [comment]
To anyone visiting this for Next v15.2.4^, adding the aforementioned variable to the `manifest.ts` file seems to build successfully.

```ts
export const dynamic = 'force-static';
```

**Reply by dell-mic** [reply]
For me this works, but only when running `next dev` _without_ ` --turbopack` option. So might be turbopack related?
